docs/source/development/cdf_guide.rst

```diff
@@ -0,0 +1,677 @@
+.. _cdf_guide:
+
+
+##############################
+Generating CDF Files
+##############################
+
+******************************
+CDF Files Introduction
+******************************
+CDF files are binary files which require special libraries to create and manipulate. Goddard Space Flight Center (GSFC) maintains these CDF libraries, and is the shepherd of the file format.
+
+The GSFC describes CDF files as a "self-describing data format for the storage of scalar and multidimensional data in a platform- and discipline-independent way".
+
+IMAP is archiving its science data with the `Space Physics Data Facility (SPDF) <https://spdf.gsfc.nasa.gov/>`_, and they require data to be in the CDF file format.
+
+CDF Ecosystem
+===============
+**CDF** refers to more than just the ".cdf" files themselves, but the entire toolkit surrounding those files.  The official toolkits are available in IDL, C, Fortran and MATLAB.
+
+For python implementations, there is **pycdf**, which is a part of **spacepy**.  Pycdf wraps the C libraries, and also includes various bells and whistles that help make the files a litle more python friendly.
+
+There is also **cdflib**, which is a pure-python implementation of the CDF libraries. **cdflib** doesn't require installing the C libraries, and also contains the functions "xarray_to_cdf" and "cdf_to_xarray" to convert between xarray Dataset objects in python and the CDF file formats.
+
+What's in a CDF file?
+======================
+
+There are two parts of a CDF file that a scientist using the files would care about: Variables and Attributes.
+
+The way these are actually stored in the file are somewhat complex, but it doesn't matter to the end user.  The CDF libraries enable users to easily list and extract the variables and attributes within the CDF files.
+
+In case you have several spare hours, the internal format of CDF files are described `here <https://cdaweb.gsfc.nasa.gov/pub/software/cdf/doc/cdf391/cdf39ifd.pdf>`_.
```

Comment:
```suggestion
In case you have several spare hours, the internal format of CDF files are described in the `CDF specification <https://cdaweb.gsfc.nasa.gov/pub/software/cdf/doc/cdf391/cdf39ifd.pdf>`_.
```